src/frontend/src/optimizer/plan_node/logical_over_window.rs

```diff
@@ -98,14 +120,214 @@ impl LogicalOverWindow {
 
         let mut window_funcs = vec![];
         for expr in &mut select_exprs {
-            if let ExprImpl::WindowFunction(_) = expr {
-                let new_expr =
-                    InputRef::new(input_len + window_funcs.len(), expr.return_type().clone())
-                        .into();
-                let f = std::mem::replace(expr, new_expr)
-                    .into_window_function()
-                    .unwrap();
-                window_funcs.push(*f);
+            if let ExprImpl::WindowFunction(window) = expr {
+                let (kind, args, return_type, partition_by, order_by, frame) = (
+                    window.kind,
+                    &window.args,
+                    &window.return_type,
+                    &window.partition_by,
+                    &window.order_by,
+                    &window.frame,
+                );
+                if let WindowFuncKind::Aggregate(agg_kind) = kind
+                    && matches!(
+                        agg_kind,
+                        AggKind::Avg
+                            | AggKind::StddevPop
+                            | AggKind::StddevSamp
+                            | AggKind::VarPop
+                            | AggKind::VarSamp
+                    )
+                {
+                    // Refer to LogicalAggBuilder::try_rewrite_agg_call()
```

Comment:
Could we extract the common logic into something like `Agg::breakdown_agg_call`?

Reply:
I think we can do it in later PR. It is a little comples